dlang/psi-impl/src/main/kotlin/io/github/intellij/dlanguage/psi/ext/DLanguageTypeImplMixin.kt

```diff
@@ -3,23 +3,15 @@ package io.github.intellij.dlanguage.psi.ext
 import com.intellij.extapi.psi.ASTWrapperPsiElement
 import com.intellij.lang.ASTNode
 import io.github.intellij.dlanguage.psi.DLanguageType
-import io.github.intellij.dlanguage.psi.types.DArrayType
-import io.github.intellij.dlanguage.psi.types.DPointerType
 import io.github.intellij.dlanguage.psi.types.DType
+import io.github.intellij.dlanguage.psi.types.getDtypeWithSuffixes
 
 abstract class DLanguageTypeImplMixin(node: ASTNode) : ASTWrapperPsiElement(node),
     DLanguageType {
 
     override fun getDType(): DType {
         // TODO type constructor
         var dtype = basicType!!.dType
-        for (suffix in typeSuffixs) {
-                if (suffix.getOP_ASTERISK() != null)
-                    dtype = DPointerType(dtype)
-                else if (suffix.getOP_BRACKET_LEFT() != null)
-                    dtype = DArrayType(dtype, null) // TODO size
-                // TODO others
-            }
-        return dtype
+        return getDtypeWithSuffixes(dtype, typeSuffixs)
```

Comment:
refactoring and improvement, use existing method to determine type suffix instead of re-implementing it.